gensim/models/phrases.py

```diff
@@ -169,8 +169,8 @@ def learn_vocab(sentences, max_vocab_size, delimiter=b'_', progress_per=10000):
             if sentence_no % progress_per == 0:
                 logger.info("PROGRESS: at sentence #%i, processed %i words and %i word types" %
                             (sentence_no, total_words, len(vocab)))
-            # sentence = [utils.any2utf8(w) for w in sentence]
-            # Unicode tokens in dictionary (not utf8)
+
+            sentence = [w for w in (utils.any2utf8('_'.join(w for w in sentence)).split('_'))]
```

Comment:
This should all be in the optimized path (C / Cython), so there's no point wasting time playing with Python calls.

The utf8 conversion overhead will be ~0, once this is optimized properly.

You can check with `cython -a` to see how much slow (Python) code is still left on the critical path. There should be none.